### ISSUE TITLE
[1.7 Dev] Openssl 1.0 does not work on Amazon Linux (or may in general)

### ISSUE THREAD

**Issue by PettitWesley** [fluent/fluent-bit#2973]
Openssl 1.0 support seems to be broken.

This can probably be tested in a container.

Install openssl1.0 with: `sudo yum install openssl-devel`

One issue I am seeing is that I can't figure out how to install openssl 1.1 on Amazon Linux.

If you want a dockerfile for Fluent Bit with amazon linux, this will work as a starting point: https://github.com/aws/aws-for-fluent-bit/blob/mainline/Dockerfile

That dockerfile will pull info from another docker image that is supposed to exist locally. So either you can create your own dockerfile based on that one, or build everything in that package with `make release`.

Openssl 1.0 seems broken for all plugins:

AWS plugins:
```
[2021/01/26 23:54:36] [ info] [output:cloudwatch_logs:cloudwatch_logs.0] Creating log group fluent
[2021/01/26 23:54:36] [debug] [upstream] connection #27 failed to logs.us-west-2.amazonaws.com:443
[2021/01/26 23:54:36] [error] [aws_client] connection initialization error
[2021/01/26 23:54:36] [error] [output:cloudwatch_logs:cloudwatch_logs.0] Failed to create log group
```

HTTP output: 
```
$ ./bin/fluent-bit -i random -p samples=1 -o http://fluentbit.io:443 -p tls=on -f 1

... 


[2021/01/26 23:55:58] [error] [output:http:http.0] no upstream connections available to fluentbit.io:443
```

Datadog:

```
[2021/01/26 23:57:18] [debug] [upstream] connection #29 failed to http-intake.logs.datadoghq.com:443
[2021/01/26 23:57:18] [debug] [out coro] cb_destroy coro_id=5
[2021/01/26 23:57:18] [debug] [task] task_id=1 reached retry-attempts limit 1/1
```

I have openssl 1.0:

```
$ rpm -qa | grep openssl
openssl-libs-1.0.2k-19.amzn2.0.4.x86_64
openssl-devel-1.0.2k-19.amzn2.0.4.x86_64
```

```
$ cat ../include/fluent-bit/flb_info.h|grep OPEN
#ifndef FLB_HAVE_OPENSSL
#define FLB_HAVE_OPENSSL
#define FLB_INFO_FLAGS " FLB_HAVE_PARSER FLB_HAVE_RECORD_ACCESSOR FLB_HAVE_STREAM_PROCESSOR JSMN_PARENT_LINKS JSMN_STRICT FLB_HAVE_TLS FLB_HAVE_OPENSSL FLB_HAVE_AWS FLB_HAVE_SIGNV4 FLB_HAVE_SQLDB FLB_HAVE_METRICS FLB_HAVE_HTTP_SERVER FLB_HAVE_SYSTEMD FLB_HAVE_FORK FLB_HAVE_TIMESPEC_GET FLB_HAVE_GMTOFF FLB_HAVE_UNIX_SOCKET FLB_HAVE_PROXY_GO FLB_HAVE_LIBBACKTRACE FLB_HAVE_REGEX FLB_HAVE_UTF8_ENCODER FLB_HAVE_LUAJIT FLB_HAVE_C_TLS FLB_HAVE_ACCEPT4 FLB_HAVE_INOTIFY"
```

**Comment by PettitWesley** [comment]
When I build it with MBEDTLS it works

**Comment by fujimotos** [comment]
@PettitWesley I borrowed my company's AWS account, and could confirm that
Fluent Bit indeed fails to establish TLS connectins on Amazon Linux.

Digging around the connection error, I noticed that Amazon Linux's system
cert folder looked like this:

```bash
[ec2-user]$ tree /etc/ssl/certs
/etc/ssl/certs
├── ca-bundle.crt
├── ca-bundle.trust.crt
├── make-dummy-cert
├── Makefile
└── renew-dummy-cert
```

and OpenSSL's documentation says:

> If CApath is not NULL, it points to a directory containing CA certificates in PEM format. The files each contain one CA certificate. The files are looked up by the CA subject name hash value, which must hence be available.
>
> https://www.openssl.org/docs/man1.0.2/man3/SSL_CTX_load_verify_locations.html

So Amazon Linux evidently does not provide system certs in a format that
`SSL_CTX_load_verify_locations()` expects. As the result, Fluent Bit was
always running with no CA trusted.

As a simple fix, I posted #2974 that teaches Fluent Bit to load `ca-bundle.crt`
on RHEL-family distributions. My expectation is that the patch should fix
the bug you reported.
